### PR TITLE
Add horovod use envs

### PR DESCRIPTION
**Pull request by makar21**

### CODE REVIEW

File: tatau_core/sessions/hvd_train.py [hunk 1]
```diff
@@ -264,12 +265,14 @@ def train(self):
...
-
-        self.eval_results = {
-            'loss': loss,
-            'acc': acc
-        }
+        test_chunk_dirs = self.test_chunk_dirs
+        if len(test_chunk_dirs):
+            loss, acc = model.eval(chunk_dirs=test_chunk_dirs, batch_size=batch_size)
+
+            self.eval_results = {
```
Comment (Yulya) [inline tatau_core/sessions/hvd_train.py]: This is very small dict, it can be placed on one line.
For me example below looks more neater and easier to read. It's just my opinion and not a must-have

self.eval_results = {'loss': loss, 'acc': acc}

File: tatau_core/sessions/hvd_train.py [hunk 2]
```diff
@@ -172,13 +172,14 @@ def process_assignment(self, assignment: TaskAssignment, *args, **kwargs):
...
-        accuracy = eval_results['acc']
+        if eval_results:
+            loss = eval_results['loss']
+            accuracy = eval_results['acc']
 
-        assignment.train_result.eval_results[str(assignment.train_data.current_iteration)] = {
-            'loss': loss,
-            'accuracy': accuracy
-        }
+            assignment.train_result.eval_results[str(assignment.train_data.current_iteration)] = {
```
Comment (Yulya) [inline tatau_core/sessions/hvd_train.py]: str(assignment.train_data.current_iteration) looks a bit ugly. Maybe have a separate variable 
current_iteration = assignment.train_data.current_iteration. 
Btw I think you do not need str here, it's ok to use integers as dict keys.

Also don't like the small dicts smashed on several lines. See my other comment.
Again, it's just IMHO

File: tatau_core/sessions/mixins.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        self._downloader = None
+
+    def set_downloader(self, downloader):
+        self._downloader = downloader
+
+    def download_all(self):
+        self._downloader.download_all()
+
+    def add_to_download_train_model(self, model_code_ipfs: str):
+        assert self._downloader and 'set_downloader must be called before'
```
Comment (Yulya) [inline tatau_core/sessions/mixins.py]: I personally don't like this assert approach.
I understand that this allows to one-line raising exception if self.downloader is  not set, but it will just raise ugly AssertionError without explanation.
I feel that this is better:
if not self._downloader:
    raise Exception('set_downloader must be called before')

File: tatau_core/sessions/mixins.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        self._downloader.add_to_download_list(file_name='model.py', multihash=model_code_ipfs)
+        return self._downloader.resolve_path('model.py')
+
+    def add_to_download_weights(self, weights_file_name: str, weights_ipfs: str):
+        assert self._downloader and 'set_downloader must be called before'
+
+        assert weights_file_name
```
Comment (Yulya) [inline tatau_core/sessions/mixins.py]: Let's maybe agree on using or not using asserts.
I just don't like it, but if we agree on using it I will stay silent )

File: tatau_core/sessions/hvd_train.py (hunk 1 above)
Comment (27tech) [reply tatau_core/sessions/hvd_train.py]: self.eval_results = dict(loss=loss, acc=acc)
